test/sqllogictest/comment.slt

```diff
@@ -243,20 +243,38 @@ CREATE TYPE int4_list AS LIST (ELEMENT TYPE = int4);
 statement ok
 COMMENT ON TYPE int4_list IS 'supercool_list';
 
+statement ok
+CREATE TYPE custom_type AS (x integer, y int4_list);
+
+statement ok
+COMMENT ON TYPE custom_type IS 'custom_type_comment';
+
+statement ok
+COMMENT ON COLUMN custom_type.x IS 'custom_type_x_comment';
+
+statement ok
+COMMENT ON COLUMN custom_type.y IS 'custom_type_y_comment';
+
 query TTT
 SELECT object_type, object_sub_id, comment FROM mz_internal.mz_comments;
```

Comment:
I tried doing `SELECT name, object_type, object_sub_id, comment FROM mz_internal.mz_comments JOIN mz_objects ON mz_comments.id = mz_objects.id;` in this file and noticed that mz_comments is a bit awkward to use since the id has to be joined against mz_objects or mz_databases depending on the object_type. This might also be a bit annoying for customers.

Reply:
@def- I am hoping that would be less awkward after https://github.com/MaterializeInc/materialize/issues/21451 if we support it in `pg_description`